core/config_manager.py

```
@@ -279,17 +341,48 @@ def save_config(
+    server_url: str,
+    path: Path | None = None,
+    profile_name: str | None = None,
+) -> None:
+    """Update only ``[server].url`` in the config file without resetting other keys."""
+    if path is None:
+        path = default_config_path(profile_name)
+
+    if path.exists():
+        try:
+            data = tomllib.loads(path.read_text(encoding="utf-8"))
+        except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)